apps/web-mzima-client/src/app/post/location-select/location-select.component.ts

```diff
@@ -32,42 +44,43 @@ import { pointIcon } from '../../core/helpers/map';
     },
   ],
 })
-export class LocationSelectComponent implements OnInit, ControlValueAccessor {
+export class LocationSelectComponent implements OnInit, AfterViewInit, OnDestroy {
   @Input() public center: LatLngLiteral;
   @Input() public zoom: number;
+  @Input() public location: LatLngLiteral;
+  @Input() public required: boolean;
+  @Input() public parent: { form: FormGroup; latFieldsKey: string };
+  @Output() locationChange = new EventEmitter();
+  public emptyFieldLat = false;
+  public emptyFieldLng = false;
+  public noLetterLat = false;
+  public noLetterLng = false;
   private map: Map;
-  postsCollection: GeoJsonPostsResponse;
-  mapLayers: any[] = [];
-  mapReady = false;
-  mapConfig: MapConfigInterface;
-  markerClusterData = new MarkerClusterGroup();
-  markerClusterOptions: MarkerClusterGroupOptions = {
+  public mapLayers: any[] = [];
+  public mapReady = false;
+  public mapConfig: MapConfigInterface;
+  public markerClusterOptions: MarkerClusterGroupOptions = {
     animate: true,
     maxClusterRadius: 50,
   };
-  mapFitToBounds: LatLngBounds;
-  fitBoundsOptions: FitBoundsOptions = {
+  public mapFitToBounds: LatLngBounds;
+  public fitBoundsOptions: FitBoundsOptions = {
     animate: true,
   };
-  mapMarker: Marker;
-  location: LatLngLiteral;
-
+  public mapMarker: Marker;
   public leafletOptions: MapOptions;
+  public disabled = false;
+  public geocoderControl: any;
+  private geocoderSubscription?: Subscription;
```

Comment:
Because we use the **@ngneat/until-destroy** library, we do not need to create additional variables to unsubscribe

Reply:
Ok.

Reply:
Done @kvengerov

Reply:
Although there is one more problem I'm trying to fix. I did a check in the `enableSubmitButtonOnGeocode()` method.
So I get this error the second time I use the geocoding search `cannot convert undefined or null to object`